lib/src/js/js.dart

```diff
@@ -44,13 +44,15 @@ final _isStrictMode = () {
 
 const bool isJS = true;
 
-bool get isNodeJs => _process?._maybeRelease?.name == 'node';
+bool get isNodeJs =>
+    _process != null && _toString(_process) == '[object process]';
 
 bool get isBrowser =>
-    !isNodeJs &&
-    _document != null &&
-    typeofEquals<Object?>(
-        getProperty<Object?>(_document!, 'querySelector'), 'function');
+    (_window != null && _toString(_window) == '[object Window]') ||
+    _isWebWorker;
+
+bool get _isWebWorker =>
+    _importScripts != null && _toString(_importScripts) == '[object Function]';
```

Comment:
These are getting complex enough that it's probably worth making them final fields so they'll only be evaluated once.